contracts/proxies/StakingProxy.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.4.24;
+
+import "../Staking.sol";
+import "./ThinProxy.sol";
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+
+
+contract StakingProxy is ThinProxy {
+    // keccak256("aragon.network.staking")
+    bytes32 internal constant IMPLEMENTATION_SLOT = 0xbd536e2e005accda865e2f0d1827f83ec8824f3ea04ecd6131b7c10058635814;
+
+    string internal constant ERROR_WRONG_IMPLEMENTATION_SLOT = "STAKING_PROXY_WRONG_IMPL_SLOT";
+
+    constructor(Staking _implementation, ERC20 _token) ThinProxy(address(_implementation)) public {
+        require(IMPLEMENTATION_SLOT == keccak256("aragon.network.staking"), ERROR_WRONG_IMPLEMENTATION_SLOT);
+
+        bytes4 selector = _implementation.initialize.selector;
+        bytes memory initializeData = abi.encodeWithSelector(selector, _token);
+        bool success = address(_implementation).delegatecall(initializeData);
+
+        assembly {
+            let output := mload(0x40)
+            mstore(0x40, add(output, returndatasize))
```

Comment:
Yes, it is the recommended way to work with it. You could avoid updating it in case you have fully control of the memory space and you can ensure there is no chance a wrong free memory pointer address will be read. 

In the `DelegateProxy` contract, as you can see, the function is `internal` which means that anyone who calls it will be sharing the same memory space, thus, everyone calling that function will have to consider that the free memory pointer hasn't been updated, which is not the recommended way of using it. I'd suggest changing it.